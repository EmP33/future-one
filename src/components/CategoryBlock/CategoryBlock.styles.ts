import styled from "styled-components";

export const Wrapper = styled.div`
  box-shadow: 0px 2px 8px rgba(41, 55, 69, 0.15);
  height: 200px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 0.5rem;
  position: relative;
  transition: all 0.2s ease;
  cursor: pointer;

  &:hover {
    background: var(--color-green);
  }
  &:hover svg {
    color: var(--color-white);
    fill: var(--color-white);
  }
  &:hover svg path {
    fill: var(--color-white);
  }

  & svg {
    font-size: 3rem;
    color: var(--color-light-gray);
  }

  & span {
    position: absolute;
    bottom: 15px;
    right: 5px;
    display: flex;
    align-items: center;
    color: var(--color-white);
    text-decoration: underline;

    & svg {
      font-size: 1.5rem;
      color: var(--color-green);
    }
  }
`;
