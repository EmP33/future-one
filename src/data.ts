export interface IQuestion {
  id: string;
  question: string;
  answer: string;
}

export interface IQuote {
  id: string;
  title: string;
  content: string;
  quoteAuthor: string;
}

export interface ICategory {
  id: number;
  iconName: string;
}

export const questions: IQuestion[] = [
  {
    id: "01",
    question: "Arcu non odio",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
  {
    id: "02",
    question: "Eget gravida cum sociis",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
  {
    id: "03",
    question: "Massa massa ultricies ",
    answer:
      "Sapien nec sagittis aliquam malesuada bibendum. Tempor nec feugiat nisl pretium.",
  },
];

export const quote: IQuote = {
  id: "q1",
  title: "Bibendum at varius vel pharetra vel turpis nunc eget lorem.",
  content:
    "Aliquam purus sit amet luctus venenatis lectus magna. Faucibus purus in massa tempor nec feugiat nisl pretium fusce. Tortor vitae purus faucibus ornare. Neque viverra justo nec ultrices. Nisl nunc mi ipsum faucibus. Tristique risus nec feugiat in fermentum. Elit duis tristique sollicitudin nibh sit amet commodo nulla.",
  quoteAuthor: "Faucibus Vitae, Office Assistant",
};

export const categories: ICategory[] = [
  { id: 0, iconName: "Lifebuoy" },
  { id: 1, iconName: "Scissors" },
  { id: 2, iconName: "Android" },
  { id: 3, iconName: "Zoom" },
];
