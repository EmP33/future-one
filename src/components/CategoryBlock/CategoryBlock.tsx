import React from "react";
import { ICategory } from "../../data";
// Icons
import { Zoom, Lifebuoy } from "../../assets";
import { AiFillAndroid } from "react-icons/ai";
import { FiScissors } from "react-icons/fi";
import { Wrapper } from "./CategoryBlock.styles";
import { BiChevronRight } from "react-icons/bi";

const CategoryBlock: React.FC<{ category: ICategory }> = ({ category }) => {
  let icon: any;
  if (category.iconName === "Lifebuoy") icon = <Lifebuoy />;
  if (category.iconName === "Scissors") icon = <FiScissors />;
  if (category.iconName === "Android") icon = <AiFillAndroid />;
  if (category.iconName === "Zoom") icon = <Zoom />;

  return (
    <Wrapper>
      {icon}
      <span>
        View details <BiChevronRight />
      </span>
    </Wrapper>
  );
};

export default CategoryBlock;
