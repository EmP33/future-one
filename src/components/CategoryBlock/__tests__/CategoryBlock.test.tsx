import CategoryBlock from "../CategoryBlock";
import ShallowRenderer from "react-test-renderer/shallow";
import { categories } from "../../../data";

describe("<CategoryBlock />", () => {
  it("should render properly", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<CategoryBlock category={categories[0]} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
