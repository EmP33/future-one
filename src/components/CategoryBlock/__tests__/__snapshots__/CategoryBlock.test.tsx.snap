// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryBlock /> should render properly 1`] = `
<styled.div>
  <HiLifebuoy />
  <span>
    View details 
    <BiChevronRight />
  </span>
</styled.div>
`;
